solutions/cf-1528A.mdx

```diff
@@ -0,0 +1,130 @@
+---
+id: cf-1528A
+source: CF
+title: Parsa's Humongous Tree
+author: Ryan Chou
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/91058)
+
+## Explanation
+
+Without loss of generality, let's assume we've already computed the best beauty values for all of vertex $u$'s neighbors.
+
+Let $\texttt{range}_u$ store $l_u$ and $r_u$ and $\texttt{adj}_u$ store the beauty values of all vertices adjacent to $u$.
+
+$\texttt{range}_u = \{1, 10\}$
+
+$\texttt{adj}_u = \{2, 3, 6, 9\}$
+
+Let's say that we select $7$, which gives us a beauty of $|7-2| + |7-3| + |7-6| + |7-9| = 12$. Can we do better than this?
+
+Intuitively, for any $X$ that we choose to be $u$'s beauty value, if there's more elements that are greater than this $X$, making $X$ as small as possible would be more optimal (increasing differences for these elements) and vice versa. Note that in a case where the number of greater beauty values is the same as the number of smaller beauty values, setting $X$ to the minimum or maximum will never decrease the answer. Moving $X$ will always increase the distance for the other end of the beauty values.
+
+For a more rigorous explanation, note that [choosing the median minimizes the sum of absolute deviation](https://math.stackexchange.com/questions/85448/why-does-the-median-minimize-ex-c), so we'd want to get as far away from the median as possible.
```

Comment:
https://usaco.guide/general/working-mdx/#info-block using the info block would be better here

Reply:
lmk if the info block's title is kinda weird, i couldn't think of a better one